tests/positive/general/baseconfigsmerging.go

```diff
@@ -0,0 +1,178 @@
+}
+
+var (
+	baseConfig = []byte(`{
+				"ignition": { "version": "3.0.0" },
+				"storage": {
+					"files": [{
+						"path": "/foo/bar",
+						"contents": { "source": "data:,base%20config%0A" }
+					}]
+				}
+			}`)
+
+	baseConfig2 = []byte(`{
+				"ignition": { "version": "3.0.0" },
+				"storage": {
+					"files": [{
+						"path": "/foo/bar2",
+						"contents": { "source": "data:,base%20config2%0A" }
+					}]
+				}
+			}`)
+	baseConfig3 = []byte(`{
+				"ignition": { "version": "3.0.0" },
+				"storage": {
+					"files": [{
+						"path": "/foo/bar2",
+						"contents": { "source": "data:,base%20config3%0A" }
+					}]
+				}
+			}`)
+	platformConfig = []byte(`{
+				"ignition": { "version": "3.0.0" },
+				"storage": {
+					"files": [{
+						"path": "/foo/bar3",
+						"contents": { "source": "data:,platform%20config%0A" }
+					}]
+				}
+			}`)
+	platformConfig2 = []byte(`{
+				"ignition": { "version": "3.0.0" },
+				"storage": {
+					"files": [{
+						"path": "/foo/bar4",
+						"contents": { "source": "data:,platform%20config2%0A" }
+					}]
+				}
+			}`)
+	platformConfig3 = []byte(`{
+				"ignition": { "version": "3.0.0" },
+				"storage": {
+					"files": [{
+						"path": "/foo/bar2",
+						"contents": { "source": "data:,platform%20config3%0A" }
+					}]
+				}
+			}`)
+)
+
+// VerifyMergingOfMultipleBaseConfigs checks if multiple
+// base configs and/or platform configs are merged in the
+// right order.
+func VerifyMergingOfMultipleBaseConfigs() types.Test {
+	name := "merge.multiple.base.configs"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := `{
+		"ignition": {"version": "$version"}
+	}`
+	configMinVersion := "3.0.0"
+	var systemFiles []types.File
+
+	systemFiles = append(systemFiles, types.File{
+		Node: types.Node{
+			Name:      "40-base.ign",
+			Directory: "base.d",
+		},
+		Contents: string(baseConfig),
+	}, types.File{
+		Node: types.Node{
+			Name:      "50-base2.ign",
+			Directory: "base.d",
+		},
+		Contents: string(baseConfig2),
+	}, types.File{
+		Node: types.Node{
+			Name:      "60-base3.ign",
+			Directory: "base.d",
+		},
+		Contents: string(baseConfig3),
+	}, types.File{
+		Node: types.Node{
+			Name:      "10-plaform.ign",
+			Directory: "base.platform.d/file",
+		},
+		Contents: string(platformConfig),
+	}, types.File{
+		Node: types.Node{
+			Name:      "20-plaform2.ign",
+			Directory: "base.platform.d/file",
+		},
+		Contents: string(platformConfig2),
+	}, types.File{
+		Node: types.Node{
+			Name:      "30-plaform3.ign",
+			Directory: "base.platform.d/file",
+		},
+		Contents: string(platformConfig3),
+	})
+	out[0].Partitions.AddFiles("ROOT", []types.File{
+		{
+			Node: types.Node{
+				Name:      "bar",
+				Directory: "foo",
+			},
+			Contents: "base config\n",
+		},
+	})
+	out[0].Partitions.AddFiles("ROOT", []types.File{
+		{
+			Node: types.Node{
+				Name:      "bar2",
+				Directory: "foo",
+			},
+			Contents: "platform config3\n",
```

Comment:
This proves that platform overrides base, but not that base configs override each other in the right order.